packages/bundler-plugin-core/src/options-mapping.ts

```diff
@@ -52,57 +57,84 @@ export type InternalIncludeEntry = RequiredInternalIncludeEntry &
+    // release injection
+    org: userOptions.org ?? process.env["SENTRY_ORG"],
+    project: userOptions.project ?? process.env["SENTRY_PROJECT"],
+    release: userOptions.release ?? process.env["SENTRY_RELEASE"] ?? "",
+
+    // These options and can also be set via env variables or config file
+    // but are only passed to Sentry CLI. They could remain undefined
+    // because the config file is only read by the CLI
+    authToken: userOptions.authToken ?? process.env["SENTRY_AUTH_TOKEN"],
+    customHeader: userOptions.customHeader ?? process.env["CUSTOM_HEADER"],
+    url: userOptions.url ?? process.env["SENTRY_URL"] ?? "https://sentry.io/",
+    vcsRemote: userOptions.vcsRemote ?? process.env["SENTRY_VCS_REMOTE"] ?? "origin",
+
+    // Options with default values
+    finalize: userOptions.finalize ?? true,
+    cleanArtifacts: userOptions.cleanArtifacts ?? false,
+    dryRun: userOptions.dryRun ?? false,
+    debug: userOptions.debug ?? false,
+    silent: userOptions.silent ?? false,
+    telemetry: userOptions.telemetry ?? true,
+    injectReleasesMap: userOptions.injectReleasesMap ?? false,
+
+    // Optional options
+    setCommits: userOptions.setCommits,
+    deploy: userOptions.deploy,
+    entries,
+    dist: userOptions.dist,
+    errorHandler: userOptions.errorHandler,
+    configFile: userOptions.configFile,
+  };
+}
+
+/**
+ * Converts the user-facing `entries` option to the internal `entries` option
+ */
+function normalizeEntries(
+  userEntries: UserOptions["entries"]
+): (string | RegExp)[] | ((filePath: string) => boolean) | undefined {
+  if (userEntries === undefined) {
+    return undefined;
+  } else if (typeof userEntries === "function" || Array.isArray(userEntries)) {
+    return userEntries;
   } else {
-    entries = [userOptions.entries];
+    return [userEntries];
   }
+}
+
+function normalizeInclude(userOptions: UserOptions): InternalIncludeEntry[] {
+  const rawUserInclude = userOptions.include;
 
   let userInclude: UserIncludeEntry[];
-  if (typeof userOptions.include === "string") {
-    userInclude = [convertIncludePathToIncludeEntry(userOptions.include)];
-  } else if (Array.isArray(userOptions.include)) {
-    userInclude = userOptions.include.map((potentialIncludeEntry) => {
+  if (typeof rawUserInclude === "string") {
+    userInclude = [convertIncludePathToIncludeEntry(rawUserInclude)];
+  } else if (Array.isArray(rawUserInclude)) {
+    userInclude = rawUserInclude.map((potentialIncludeEntry) => {
       if (typeof potentialIncludeEntry === "string") {
         return convertIncludePathToIncludeEntry(potentialIncludeEntry);
       } else {
         return potentialIncludeEntry;
       }
     });
   } else {
-    userInclude = [userOptions.include];
+    userInclude = [rawUserInclude];
   }
 
-  const include = userInclude.map((userIncludeEntry) =>
+  return userInclude.map((userIncludeEntry) =>
     normalizeIncludeEntry(userOptions, userIncludeEntry)
   );
```

Comment:
M: If you were to use a helper like [the SDK's `arrayify` function](https://github.com/getsentry/sentry-javascript/blob/c59c70bfc636de9bd7512f37e9644db1947a5563/packages/utils/src/misc.ts#L203-L211), you could simplify this a lot, I think, to basically just be something like

```js
return arrayify(userOptions.include)
  .map((potentialIncludeEntry) =>
    typeof potentialUserIncludeEntry === "string"
      ? convertIncludePathToIncludeEntry(potentialIncludeEntry)
      : potentialIncludeEntry
  )
  .map((userIncludeEntry) =>
    normalizeIncludeEntry(userOptions, userIncludeEntry)
  );
```

Reply:
Much cleaner, thanks for the suggestion! 
[081d1de](https://github.com/getsentry/sentry-javascript-bundler-plugins/pull/103/commits/081d1de7c5991eb2e8970edc2bd5dd472c72096e)